### PR TITLE
Single Headed Attention RNN: Stop Thinking With Your Head

### PR DESCRIPTION
**Pull request by AntoineTheb**

### CODE REVIEW

File: article/_posts/2019-11-27-SHA_RNN.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+- Simple RNN architecture provides (almost) state of the art results
+- "Tokenization attacks" make comparing results difficult
+
+# Preface
+
+The currently reviewed article requires a very good understanding of NLP, attention and the transformer module. These two articles ([1.](https://jalammar.github.io/visualizing-neural-machine-translation-mechanics-of-seq2seq-models-with-attention/) [2.](http://jalammar.github.io/illustrated-transformer/)) provide a good summary of these concepts.
+
+# Introduction
+
+State of the art (SotA) NLP architectures are massive and require enormous resources to train, which makes meaningful research hard to do for smaller research teams. The author painfully introduces its own NLP architecture, based on a previous paper of his <sup>[1]</sup>, called SHA-RNN.
```
Comment (lemairecarl) [inline article/_posts/2019-11-27-SHA_RNN.md]: his own

File: article/_posts/2019-11-27-SHA_RNN.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+![](/article/images/sha-rnn/fig4.jpeg)
+
+For BPC, lower is better.
+
+## Tokenization attacks
+
+All NLP models use tokenization to simplify their input, which makes sense because you don't want to have your softmax classifier to give a probability for every word in the dictionary and more. Therefore, tokenization algorithms will break words into, usually, prefixes and suffixes, common letter grouping, etc. Any good tokenizer will give you the option of breaking your dictionary into the amount of token that you want, and an extreme example would be to have your dataset broken into tokens representing the ASCII table.
+
+However, the author mentions that breaking your dataset into fewer tokens actually makes thing more easy. Indeed, while training your model, you have much more feedback if you apply the loss and do teacher forcing on a letter-per-letter case instead of a word-per-word case. The author does not provide any experiment results backing up these claims.
```
Comment (lemairecarl) [inline article/_posts/2019-11-27-SHA_RNN.md]: makes things easier?

File: article/_posts/2019-11-27-SHA_RNN.md (hunk 1 above)
Comment (lemairecarl) [reply article/_posts/2019-11-27-SHA_RNN.md]: did you know that our markdown flavor supports references? I just added an example in the template on master

**Comment by lemairecarl** [comment]
If you merge master into your branch, the checks will run correctly

File: article/_posts/2019-11-27-SHA_RNN.md (hunk 2 above)
Comment (AntoineTheb) [reply article/_posts/2019-11-27-SHA_RNN.md]: Fixed

File: article/_posts/2019-11-27-SHA_RNN.md (hunk 1 above)
Comment (AntoineTheb) [reply article/_posts/2019-11-27-SHA_RNN.md]: Fixed it, thanks !

**Pull request merged**